src/DotNetty.Buffers/ReadOnlyByteBufferStream.cs

```diff
@@ -29,7 +29,7 @@ public ReadOnlyByteBufferStream(IByteBuffer buffer, bool releaseReferenceOnClosu
 
         public override long Length
         {
-            get { throw new NotSupportedException(); }
+            get => this.buffer.ReadableBytes;
```

Comment:
IByteBuffer.ReadableBytes may not equals to Stream.Length since it decrease after read.